ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -129,22 +127,6 @@ export default class PermissionConnect extends Component {
     ///: END:ONLY_INCLUDE_IF
   };
 
-  beforeUnload = () => {
-    const { permissionsRequestId, rejectPermissionsRequest } = this.props;
-    const { permissionsApproved } = this.state;
-
-    if (permissionsApproved === null && permissionsRequestId) {
-      rejectPermissionsRequest(permissionsRequestId);
-    }
-  };
-
-  removeBeforeUnload = () => {
-    const environmentType = getEnvironmentType();
-    if (environmentType === ENVIRONMENT_TYPE_NOTIFICATION) {
-      window.removeEventListener('beforeunload', this.beforeUnload);
-    }
-  };
```

Comment:
are we sure removing this doesnt break anything?

Reply:
I am reasonably confident this is OK to remove. This is in the permission connect flow, which is driven by ApprovalController state. And we have code in the background to detect popup closures and reject all pending approvals. So this should be caught already.

Plus we found that these beforeUnload scripts didn't work reliably anyway, so if we did need this for some reason, I'd expect it to occasionally fail. At least now it would fail consistently 😅